File: frontend/internshop-frontend/src/models/Ad.ts
import { User } from "./User";

export type Ad = {
  id?: number | null;
  title?: string;
  description?: string;
  imageUrl?: string;
  price?: number;
  category?: string;
  city?: string;
  postedDate?: string | null;
  user?: Partial<User> | null;
};

File: backend/src/main/java/com/internshop/model/Ad.java
package com.internshop.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "ads")
public class Ad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(length = 1000)
    private String description;

    private String imageUrl;

    private double price;

    private String city;

    private LocalDate postedDate;

    @Enumerated(EnumType.STRING)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false)
    private boolean isActive = true;

}
